app/v2/controllers/V2DeparturesController.scala

```diff
@@ -159,21 +163,42 @@ class V2DeparturesControllerImpl @Inject() (
         .asPresentation
     } yield declarationResult
 
-  def getMessage(departureId: MovementId, messageId: MessageId): Action[AnyContent] =
+  def getMessage(movementId: MovementId, messageId: MessageId): Action[AnyContent] =
     authActionNewEnrolmentOnly.async {
       implicit request =>
         implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
 
-        departuresService
-          .getMessage(request.eoriNumber, departureId, messageId)
-          .asPresentation
-          .fold(
-            presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
-            response => Ok(Json.toJson(HateoasDepartureMessageResponse(departureId, messageId, response)))
-          )
-
+        acceptHeaderRoute {
+          case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON)     => getMessageJson(movementId, messageId)
+          case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON_XML) => getMessageJsonWrappedXml(movementId, messageId)
+        }
     }
 
+  def getMessageJsonWrappedXml(movementId: MovementId, messageId: MessageId)(implicit hc: HeaderCarrier, request: AuthenticatedRequest[_]): Future[Result] =
+    (for {
+      messageSummary <- departuresService.getMessage(request.eoriNumber, movementId, messageId).asPresentation
+    } yield messageSummary).fold(
+      presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
+      response => Ok(Json.toJson(HateoasDepartureMessageResponse(movementId, messageId, response)))
+    )
```

Comment:
This is slightly nitpicky, but the bracket surrouding the for comprehension is redundant. Normally I wouldn't really bat an eyelid at this, but the way the brackets are laid out, I spent 15 seconds trying to work out how this worked without one more bracket -- the bracket that matches the fold is on line 182!

That said, we don't need a for comprehension for this:

```suggestion
    departuresService
        .getMessage(request.eoriNumber, movementId, messageId)
        .asPresentation
        .fold(
            presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
            response => Ok(Json.toJson(HateoasDepartureMessageResponse(movementId, messageId, response))
        )
```

Reply:
(What you've written is fine, actually, but the bracket irked me, apparently!)